examples/ddp_00.py

```diff
@@ -0,0 +1,50 @@
+from dataclasses import dataclass, field
+import logging
+import os
+from typing import Any, List
+
+import hydra
+from hydra.core.config_store import ConfigStore
+from omegaconf import DictConfig, MISSING
+import torch
+import torch.distributed as dist
+
+log = logging.getLogger(__name__)
+
+
+defaults = [
+    {"hydra/launcher": "joblib"}
+]
+
+@dataclass
+class DDPConf:
+    defaults: List[Any] = field(default_factory=lambda: defaults)
+    backend: str = "gloo"
+    init_method: str = MISSING
```

Comment:
```suggestion
    MASTER_ADDR: str = "127.0.0.1"
    MASTER_PORT: str = str(random.randint(48620,49150))
    init_method: str = "tcp://" + MASTER_ADDR + ":" + MASTER_PORT
```

Howabout something like this?

Reply:
What happens if you override the MASTER_PORT externally?
(my guess is that it will not work correctly)

Reply:
What do you mean, externally? As in change the os environment variable? I'm not sure. I think they recommend either handling this fully within `init_method` or by setting ENV variables, but not both.